lib/dry/validation/result.rb

```diff
@@ -81,3 +81,11 @@ def result_ast
     end
   end
 end
+
+begin
```

Comment:
could we add it to something reusable like `Dry::Validation.load_extensions(:monads, :struct)` or something like that? this way we'd reduce duplication and be able to easily have tests for it